src/mod_vcard.erl

```diff
@@ -196,6 +204,114 @@ stop(VHost) ->
     gen_server:call(Proc, stop),
     ejabberd_sup:stop_child(Proc).
 
+%%--------------------------------------------------------------------
+%% config_spec
+%%--------------------------------------------------------------------
+
+-spec config_spec() -> mongoose_config_spec:config_section().
+config_spec() ->
+    #section{
+       items = #{<<"iqdisc">> => mongoose_config_spec:iqdisc(),
+                 <<"host">> => #option{type = string,
+                                       validate = non_empty},
+                 <<"search">> => #option{type = boolean},
+                 <<"backend">> => #option{type = atom,
+                                          validate = {module, mod_vcard}},
+                 <<"matches">> => #option{type = int_or_infinity,
+                                          validate = non_negative},
+                 <<"ldap_pool_tag">> => #option{type = atom,
+                                                validate = non_empty},
+                 <<"ldap_base">> => #option{type = string},
+                 <<"ldap_uids">> => #list{items = ldap_uids_spec()},
+                 <<"ldap_filter">> => #option{type = string},
+                 <<"ldap_deref">> => #option{type = atom,
+                                             validate = {enum, [never, always, finding, searching]}},
+                 <<"ldap_vcard_map">> => #list{items = ldap_vcard_map_spec()},
+                 <<"ldap_search_fields">> => #list{items = ldap_search_fields_spec()},
+                 <<"ldap_search_reported">> => #list{items = ldap_search_reported_spec()},
+                 <<"ldap_search_operator">> => #option{type = atom,
+                                                       validate = {enum, ['or', 'and']}},
+                 <<"ldap_binary_search_fields">> => #list{items = #option{type = binary,
+                                                                         validate = non_empty}},
+                 <<"riak">> => riak_config_spec()
+                }
+      }.
+
+ldap_uids_spec() ->
+    #section{
+        items = #{<<"attr">> => #option{type = string},
+                <<"format">> => #option{type = string}},
```

Comment:
Minor: indentation

Reply:
Will reuse the one from `mongoose_config_spec:ldap_uids()` and have removed `ldap_uids_spec()` from `mod_vcard`.